src/webapp/components/settings-form/SettingsForm.tsx

```diff
@@ -0,0 +1,357 @@
+                <Button
+                    disabled={loading}
+                    color="primary"
+                    onClick={saveSettings}
+                    variant="contained"
+                    type="submit"
+                >
+                    {i18n.t("Save")}
+                </Button>
+            </DialogActions>
+        </Dialog>
+    );
+});
+
+type GeneralFormProps = {
+    appSettings: AppSettings;
+    onChange: (appSettings: AppSettings) => void;
+    settingsData: AppSettingsData;
+};
+
+const GeneralForm = (props: GeneralFormProps) => {
+    const { appSettings, onChange, settingsData } = props;
+
+    const updateProject = (item: Maybe<string | number>, field: keyof AppSettings) => {
+        onChange(AppSettings.create({ ...appSettings, [field]: item }));
+    };
+
+    const monthsItems = getMonths();
+    const unitsItems = getUnits();
+    const daysPerMonthYear = getDaysPerMonthYear(appSettings.periodEndDateMonth);
+
+    return (
+        <GeneralFormContainer>
+            <ModalSelector
+                items={settingsData.categories}
+                value={appSettings.defaultProjectId}
+                onChange={item => updateProject(item?.value, "defaultProjectId")}
+                showEmptyValue
+                label={i18n.t("Projects Category")}
+            />
+
+            <ModalSelector
+                items={settingsData.combinations}
+                value={appSettings.categoryComboId}
+                onChange={item => updateProject(item?.value, "categoryComboId")}
+                showEmptyValue
+                label={i18n.t("Category combination")}
+            />
+
+            <ModalSelector
+                items={settingsData.dataElementsGroupSets}
+                value={appSettings.coreCompetencyId}
+                onChange={item => updateProject(item?.value, "coreCompetencyId")}
+                showEmptyValue
+                label={i18n.t("Core competency data element group set")}
+            />
+
+            <div style={{ display: "flex" }}>
+                <div style={{ flex: "1" }}>
+                    <Dropdown
+                        className="dropdown dropdown-resetmargins"
+                        hideEmpty
+                        items={monthsItems}
+                        label={i18n.t("Default Period End Date - Month")}
+                        onChange={value => updateProject(Number(value), "periodEndDateMonth")}
+                        value={String(appSettings.periodEndDateMonth)}
+                    />
+                </div>
+                <Dropdown
+                    hideEmpty
+                    items={daysPerMonthYear}
+                    onChange={value => updateProject(Number(value), "periodEndDateDay")}
+                    label={i18n.t("Day")}
+                    value={String(appSettings.periodEndDateDay)}
+                />
+            </div>
+            <div style={{ display: "flex" }}>
+                <div style={{ flex: "1" }}>
```

Comment:
move to some styles to avoid that this hardcoded styles re-render full components